pkg/api/validation/validation.go

```diff
@@ -1803,6 +1803,18 @@ func validateTolerations(tolerations []api.Toleration, fldPath *field.Path) fiel
 		// validate the toleration key
 		allErrors = append(allErrors, unversionedvalidation.ValidateLabelName(toleration.Key, idxPath.Child("key"))...)
 
+		if toleration.ForgivenessSeconds != nil {
+			if *toleration.ForgivenessSeconds <= 0 {
+				allErrors = append(allErrors, field.Invalid(idxPath.Child("forgivenessSeconds"), toleration.ForgivenessSeconds, "forgivenessSeconds must be greater than zero when set"))
+			}
+			if toleration.Effect != api.TaintEffectNoExecute {
+				allErrors = append(allErrors, field.Invalid(idxPath.Child("effect"), toleration.Effect, "effect must be NoExecute when `ForgivenessSeconds` is set"))
+			}
+			if toleration.Operator != api.TolerationOpExists {
+				allErrors = append(allErrors, field.Invalid(idxPath.Child("operator"), toleration.Operator, "operator must be Exists when `key` is 'nodeNotReady'"))
```

Comment:
I don't think there is any particular reason, just in node notready/unreachable case, taints has no value, what about removing this checking?